model/textparse/openmetricsparse.go

```diff
@@ -511,6 +622,21 @@ func (p *OpenMetricsParser) parseMetricSuffix(t token) (Entry, error) {
 			return EntryInvalid, p.parseError("expected next entry after timestamp", t3)
 		}
 	}
+
+	if p.skipCTSeries {
```

Comment:
Commented here before, maybe you answered and I didn't notice? 

This ideally goes level up. It's weird we do p.Next in some lover level helper for parsing metric suffixes. 

![image](https://github.com/user-attachments/assets/0351e1e7-eb68-4166-8352-5c89f9692479)

(maybe not spaghetti, but close)

Reply:
Leave for future PR? 😁 
I am tracking but iirc we discussed it's something that can be addressed after this one's done. I might be wrong tho

Reply:
refactored, let me know what you think